memory_core/memory_core_magma.py

```diff
@@ -120,7 +120,7 @@ def __init__(self,
         self.add_clk_enable = add_clk_enable
         self.add_flush = add_flush
         self.gen_addr = gen_addr
-        self.tech_map = tech_map
+        self.tech_map = GF_Tech_Map(depth=self.mem_depth, width=self.mem_width, dual_port=True)
```

Comment:
Don't change this line, just capture the tech map in the constructor. You can set the memory width different than the tech map memory width and construct a memory out of multiple tech map macros.

Reply:
Thanks!